DxDispatch/src/dxdispatch/DmlDispatchable.cpp

```diff
@@ -72,30 +101,43 @@ void FillBindingData(
         {
             auto& sources = bindingIterator->second;
 
-            if (bindPoints[i].resourceCount != sources.size())
-            {
-                throw std::invalid_argument(fmt::format(
-                    "Bind point '{}' requires {} resources, but {} were bound.", 
-                    bindPointName, 
-                    bindPoints[i].resourceCount, 
-                    sources.size()));
-            }
-
             for (auto& source : sources)
             {
                 assert(source.resource != nullptr);
-                assert(source.resourceDesc != nullptr);
-
-                if (!std::holds_alternative<Model::BufferDesc>(source.resourceDesc->value))
+                if (isSerializedGraph)
                 {
-                    throw std::invalid_argument("DML operators only support buffer bindings");
+                    // Use SafeMultiply and CalculateSize only for DMLSerialized
+                    uint64_t offset = SafeMultiply(source.elementOffset, source.elementSizeInBytes);
+                    uint64_t sizeInBytes = CalculateSize(source.elementCount, source.elementSizeInBytes);
+
+                    bindingData.bufferBindings[bufferIndex].Offset = offset;
+                    bindingData.bufferBindings[bufferIndex].SizeInBytes = sizeInBytes - offset;
+
+                    // Validation
+                    if (offset + SafeMultiply(source.elementCount, source.elementSizeInBytes) > sizeInBytes)
+                    {
+                        throw std::invalid_argument(fmt::format(
+                            "Buffer size ({} bytes) is too small for the data ({} bytes) at offset {} bytes for binding point '{}'", 
+                            sizeInBytes,
+                            SafeMultiply(source.elementCount, source.elementSizeInBytes),
+                            offset,
+                            bindPointName));
+                    }
+                }
+                else
+                {
+                    assert(source.resourceDesc != nullptr);
+                    
+                    if (!std::holds_alternative<Model::BufferDesc>(source.resourceDesc->value))
+                    {
+                        throw std::invalid_argument("DML operators only support buffer bindings");
+                    }
+
+                    auto& bufferDesc = std::get<Model::BufferDesc>(source.resourceDesc->value);
+                    bindingData.bufferBindings[bufferIndex].SizeInBytes = bufferDesc.sizeInBytes - bindingData.bufferBindings[bufferIndex].Offset;
```

Comment:
>bindingData.bufferBindings[bufferIndex].Offset [](http://example.com/codeflow?start=99&length=46)

Where is this value gets populated? Right now it seems like it will pick undefined value.